### PR TITLE
Replace TryParse with Parse

### PR DESCRIPTION
**Pull request by FirestarJes**
## Description

We have replaced all instances of **TryParse** with **Parse** and throws exceptions instead if a parsing error occurs

## References
* #228

### CODE REVIEW

File: source/PostOffice.Communicator.Tests/Dequeue/DequeueNotificationParserTests.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            // act
+            var actual = target.Parse(validBytes);
+
+            // assert
+            Assert.NotNull(actual);
+            Assert.Equal("06FD1AB3-D650-45BC-860E-EE598A3623CA", actual.Recipient);
+        }
+
+        [Fact]
+        public void Parse_BytesInvalidValid_Throws_Exception()
```
Comment (defectiveAi) [inline source/PostOffice.Communicator.Tests/Dequeue/DequeueNotificationParserTests.cs:48]: BytesInvalidValid. Proposal, SchroedingerBytes.

Comment (FirestarJes) [reply source/PostOffice.Communicator.Tests/Dequeue/DequeueNotificationParserTests.cs:48]: Proposal Accepted

Comment (FirestarJes) [reply source/PostOffice.Communicator.Tests/Dequeue/DequeueNotificationParserTests.cs:48]: Fixed

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Dequeue/DequeueNotificationParser.cs [hunk 2]
```diff
@@ -12,20 +12,30 @@
...
 using GreenEnergyHub.PostOffice.Communicator.Contracts;
+using GreenEnergyHub.PostOffice.Communicator.Exceptions;
 using GreenEnergyHub.PostOffice.Communicator.Model;
 
 namespace GreenEnergyHub.PostOffice.Communicator.Dequeue
 {
     public class DequeueNotificationParser : IDequeueNotificationParser
     {
-        public bool TryParse(byte[] dequeueNotificationContract, out DequeueNotificationDto dequeueNotificationDto)
+#pragma warning disable CA1822
```
Comment (defectiveAi) [inline source/PostOffice.Communicator/source/PostOffice.Communicator/Dequeue/DequeueNotificationParser.cs:24]: Pragma not needed, if the interface has been updated.

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs [hunk 3]
```diff
@@ -77,11 +76,15 @@ public async ValueTask DisposeAsync()
...
             if (response == null)
                 return null;
-
-            if (!_responseBundleParser.TryParse(response.Body.ToArray(), out var dto))
+            try
+            {
+                var dataBundleResponseDto = _responseBundleParser.Parse(response.Body.ToArray());
+                return dataBundleResponseDto;
+            }
+            catch (PostOfficeCommunicatorException)
```
Comment (defectiveAi) [inline source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs:84]: I think it is fine to just let PostOfficeCommunicatorException bubble up. The post-office/sub-domains already know of it, and it is more expressive than InvalidOpException.

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/RequestBundleParser.cs [hunk 4]
```diff
@@ -13,49 +13,43 @@
...
             try
             {
-                var message = new RequestBundleRequest { IdempotencyId = request.IdempotencyId, UUID = { request.DataAvailableNotificationIds } };
-                bytes = message.ToByteArray();
+               var message = new RequestBundleRequest { IdempotencyId = request.IdempotencyId, UUID = { request.DataAvailableNotificationIds } };
+               return message.ToByteArray();
             }
 #pragma warning disable CA1031
-            catch (Exception)
+            catch (Exception e)
```
Comment (defectiveAi) [inline source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/RequestBundleParser.cs:35]: You can remove try..catch. There is no protobuf ParseFrom here, so no exception can ever occur.

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs [hunk 5]
```diff
@@ -13,55 +13,61 @@
...
             if (!requestDataBundleResponseDto.IsErrorResponse)
             {
-                contract.Success = new RequestBundleResponse.Types.FileResource { Uri = requestDataBundleResponseDto.ContentUri.AbsoluteUri };
-                bytes = contract.ToByteArray();
+                try
+                {
+                    contract.Success = new RequestBundleResponse.Types.FileResource { Uri = requestDataBundleResponseDto.ContentUri.AbsoluteUri };
+                    return contract.ToByteArray();
+                }
+                catch (Exception e)
```
Comment (defectiveAi) [inline source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs:38]: Same here, no need to try...catch.

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs [hunk 6]
```diff
@@ -13,55 +13,61 @@
...
-                var contractErrorReason = MapToFailureReason(requestDataBundleResponseDto.ResponseError.Reason);
-                contract.Failure = new RequestBundleResponse.Types.RequestFailure { Reason = contractErrorReason, FailureDescription = requestDataBundleResponseDto.ResponseError.FailureDescription };
-                bytes = contract.ToByteArray();
+                try
+                {
+                    var contractErrorReason = MapToFailureReason(requestDataBundleResponseDto.ResponseError.Reason);
+                    contract.Failure = new RequestBundleResponse.Types.RequestFailure { Reason = contractErrorReason, FailureDescription = requestDataBundleResponseDto.ResponseError.FailureDescription };
+                    return contract.ToByteArray();
+                }
+                catch (Exception e)
```
Comment (defectiveAi) [inline source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs:51]: Same here, no need to try...catch.

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs (hunk 3 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs:84]: Ok, ill fix that

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Dequeue/DequeueNotificationParser.cs (hunk 2 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Dequeue/DequeueNotificationParser.cs:24]: Removed

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs (hunk 3 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/DataBundleRequestSender.cs:84]: fixed

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs (hunk 5 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs:38]: Removed

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs (hunk 6 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/ResponseBundleParser.cs:51]: Removed

File: source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/RequestBundleParser.cs (hunk 4 above)
Comment (FirestarJes) [reply source/PostOffice.Communicator/source/PostOffice.Communicator/Peek/RequestBundleParser.cs:35]: Removed

**Review by defectiveAi — APPROVED** [review]

**Review by mbusk88 — APPROVED** [review]
Nice

**Pull request merged**